lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -5071,7 +5071,10 @@ class ConvertAtenFillScalarOp : public OpConversionPattern<AtenOpT> {
       return rewriter.notifyMatchFailure(
           op, "Supplied value must be a Scalar constant");
 
-    rewriter.replaceOpWithNewOp<tosa::CastOp>(op, outType, constOp);
+    if (outType != constOp.getType())
+      rewriter.replaceOpWithNewOp<tosa::CastOp>(op, outType, constOp);
```

Comment:
fyi, you can do rewriter.createOrFold, which would return the input in case the new op can get folded

Reply:
Good to know. I'll change it. Thanks!